drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
 	}
 
-	/* Set MSIS Range */
-	LL_RCC_MSIS_Enable();
-
-	/* Wait till MSIS is ready */
-	while (LL_RCC_MSIS_IsReady() != 1) {
+	r = get_vco_input_range(STM32_PLL_M_DIVISOR, &vco_input_range);
+	if (r < 0) {
+		return r;
 	}
 
-#endif /* STM32_SYSCLK_SRC_MSIS || STM32_PLL_SRC_MSIS */
-}
+	LL_RCC_PLL1_SetDivider(STM32_PLL_M_DIVISOR);
 
-/*
- * Configure PLL as source of SYSCLK
- */
-void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
-{
-#ifdef STM32_SYSCLK_SRC_PLL
-	LL_UTILS_PLLInitTypeDef s_PLLInitStruct;
+	LL_RCC_PLL1_SetVCOInputRange(vco_input_range);
 
-	/* configure PLL input settings */
-	config_pll_init(&s_PLLInitStruct);
+	LL_RCC_PLL1_SetN(STM32_PLL_N_MULTIPLIER);
 
-	/*
-	 * Switch to HSI and disable the PLL before configuration.
-	 * (Switching to HSI makes sure we have a SYSCLK source in
-	 * case we're currently running from the PLL we're about to
-	 * turn off and reconfigure.)
-	 *
-	 * Don't use s_ClkInitStruct.AHBCLKDivider as the AHB
-	 * prescaler here. In this configuration, that's the value to
-	 * use when the SYSCLK source is the PLL, not HSI.
-	 */
-	clock_switch_to_hsi(LL_RCC_SYSCLK_DIV_1);
-	LL_RCC_PLL1_Disable();
+	/* FRACN disable DIVP,DIVQ,DIVR enable*/
+	LL_RCC_PLL1FRACN_Disable();
 
-	if (IS_ENABLED(STM32_PLL_Q_DIVISOR)) {
-		LL_RCC_PLL1_SetQ(STM32_PLL_Q_DIVISOR);
+	if (IS_ENABLED(STM32_PLL_P_ENABLED)) {
+		LL_RCC_PLL1_SetP(STM32_PLL_P_DIVISOR);
+		LL_RCC_PLL1_EnableDomain_SAI();
 	}
 
-	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
-
-	if (IS_ENABLED(STM32_PLL_SRC_MSIS)) {
-		set_up_clk_msis();
-
-		/* Switch to PLL with MSI as clock source */
-		LL_PLL_ConfigSystemClock_MSI(&s_PLLInitStruct, &s_ClkInitStruct);
-
-		/* Disable other clocks */
-		LL_RCC_HSI_Disable();
-		LL_RCC_HSE_Disable();
-
-	} else if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
-		/* Switch to PLL with HSI as clock source */
-		LL_PLL_ConfigSystemClock_HSI(&s_PLLInitStruct, &s_ClkInitStruct);
-
-		/* Disable other clocks */
-		LL_RCC_HSE_Disable();
-		LL_RCC_MSIS_Disable();
+	if (IS_ENABLED(STM32_PLL_Q_ENABLED)) {
+		LL_RCC_PLL1_SetQ(STM32_PLL_Q_DIVISOR);
+		LL_RCC_PLL1_EnableDomain_48M();
+	}
 
-	} else if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
-		int hse_bypass;
+	if (IS_ENABLED(STM32_PLL_R_ENABLED)) {
+		LL_RCC_PLL1_SetR(STM32_PLL_R_DIVISOR);
+		LL_RCC_PLL1_EnableDomain_SYS();
+	}
 
-		if (IS_ENABLED(STM32_HSE_BYPASS)) {
-			hse_bypass = LL_UTILS_HSEBYPASS_ON;
-		} else {
-			hse_bypass = LL_UTILS_HSEBYPASS_OFF;
-		}
+	LL_RCC_PLL1_Enable();
+	while (LL_RCC_PLL1_IsReady() != 1U) {
+	}
 
-		/* Switch to PLL with HSE as clock source */
-		LL_PLL1_ConfigSystemClock_HSE(CONFIG_CLOCK_STM32_HSE_CLOCK,
-					      hse_bypass,
-					      &s_PLLInitStruct,
-					      &s_ClkInitStruct);
+#else
+	/* Init PLL source to None */
+	LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_NONE);
 
-		/* Disable other clocks */
-		LL_RCC_HSI_Disable();
-		LL_RCC_MSIS_Disable();
-	}
+#endif /* STM32_PLL_ENABLED */
 
-#endif /* STM32_SYSCLK_SRC_PLL */
+	return 0;
```

Comment:
` return; `
--> void config_src_sysclk_pll

Reply:
I think the change on this file is too big and the diff is misleading. 
Function `config_src_sysclk_pll` has been removed